.github/workflows/publish.yml

```diff
@@ -17,81 +17,23 @@ on:
-              echo "Error: custom_version must match semantic versioning (e.g., v1.2.3)."
-              exit 1
-            fi
-            echo "::set-output name=release_version::${{ github.event.inputs.custom_version }}"
-          elif [[ "${GITHUB_REF}" == refs/tags/* ]]; then
-            echo "Using tag reference: ${GITHUB_REF#refs/tags/}"
-            echo "::set-output name=release_version::${GITHUB_REF#refs/tags/}"
-          else
-            echo "Defaulting to branch name: ${GITHUB_REF#refs/heads/}"
-            echo "::set-output name=release_version::${GITHUB_REF#refs/heads/}"
-          fi
-      - name: Get release version without prefix
-        id: get_release_version_without_prefix
-        env:
-          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
-        run: |
-          if [[ "${RELEASE_VERSION}" == v* ]]; then
-            echo "::set-output name=version_without_prefix::${RELEASE_VERSION:1}"
-          else
-            echo "::set-output name=version_without_prefix::${RELEASE_VERSION}"
-          fi
-      - name: Get highest semver
-        id: get_highest_semver
-        env:
-          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
-        run: |
-          if [[ -n "${{ github.event.inputs.custom_version }}" ]]; then
-            HIGHEST_SEMVER_TAG="${{ github.event.inputs.custom_version }}"
-            echo "::set-output name=highest_semver_tag::$HIGHEST_SEMVER_TAG"
-            echo "Using custom version as highest semantic version: $HIGHEST_SEMVER_TAG"
-          else 
-            source infra/scripts/setup-common-functions.sh
-            SEMVER_REGEX='^v[0-9]+\.[0-9]+\.[0-9]+(-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?$'
-            if echo "${RELEASE_VERSION}" | grep -P "$SEMVER_REGEX" &>/dev/null ; then
-              echo ::set-output name=highest_semver_tag::$(get_tag_release -m)
-              echo "Using infra/scripts/setup-common-functions.sh to generate highest semantic version: $HIGHEST_SEMVER_TAG"
-            fi
-          fi
-      - name: Check output
-        env:
-          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
-          VERSION_WITHOUT_PREFIX: ${{ steps.get_release_version_without_prefix.outputs.version_without_prefix }}
-          HIGHEST_SEMVER_TAG: ${{ steps.get_highest_semver.outputs.highest_semver_tag }}
-        run: |
-          echo $RELEASE_VERSION
-          echo $VERSION_WITHOUT_PREFIX
-          echo $HIGHEST_SEMVER_TAG
+    uses: ./.github/workflows/get_semantic_release_version.yaml
+    with:
+      custom_version: ${{ github.event.inputs.custom_version }}
+      token: ${{ github.event.inputs.token }}
+
+  publish-python-sdk:
+    uses: ./.github/workflows/publish_python_sdk.yaml
+    with:
+      custom_version: ${{ github.event.inputs.custom_version }}
+      token: ${{ github.event.inputs.token }}
 
   build-publish-docker-images:
```

Comment:
is this section redundant since its also being done in the `publish-images.yml`?

Reply:
So `publish.yml` is meant to orchestrate the whole end-to-end workflow. `publish-images.yml` is just the workflow it is calling.

I separated them out so that each one can be triggered manually for testing if necessary. That'll help maintainability and debuggability.

Let me know if that doesn't make sense though, not a complete expert on github actions.